src/desktop/main.cc

```diff
@@ -610,13 +613,140 @@ MAIN {
       return;
     }
 
-    if (message.name == "show") {
-      auto windowIndexToShow = std::stoi(message.get("window"));
-      windowIndexToShow = windowIndexToShow < 0 ? 0 : windowIndexToShow;
+    if (message.name == "getWindows") {
+      const auto index = message.index;
+      const auto showTitle = message.get("title") == "true" ? true : false;
+      const auto showSize = message.get("size") == "true" ? true : false;
+      const auto showStatus = message.get("status") == "true" ? true : false;
+      const auto window = windowManager.getWindow(index);
+      StringStream ss;
+      bool first = true;
+      ss << "[";
+      for (auto windowWithMetadata : windowManager.windows) {
```

Comment:
maybe we can add a `json()` method to the `ManagedWindow` class that returns a `JSON::Object` and a `json()` to `WindowManager` that returns an `JSON::Array` of those objects

Reply:
that's a good idea!

Reply:
@jwerle how can I implement these conditions using `JSON::Object`?

```js
            if (showTitle) {
               const auto title = w->getTitle();
               ss << "\"title\":\"" << title << "\",";
             }
             if (showSize) {
               const auto size = w->getSize();
               ss << "\"width\":" << size.width << ",";
               ss << "\"height\":" << size.height << ",";
             }
             if (showStatus) {
               const auto status = windowWithMetadata->status;
               ss << "\"status\":" << status << ",";
             }
             ss << "\"index\":" << windowWithMetadata->opts.index;
             ss << "}";
```

Reply:
aha, Object::Entries is `std::map`!

Reply:
done